content/api/commands/origin.md

```diff
@@ -0,0 +1,482 @@
+[discouraged](/guides/references/best-practices#Having-tests-rely-on-the-state-of-previous-tests)
+we know some users have historically written tests this way, often to get around
+the same-origin restrictions. But with `cy.origin()` you no longer need these
+kind of brittle hacks, as your multi-origin logic can all reside in a single
+test, like the following.
+
+<Badge type="success">After</Badge> One big test using `cy.origin()`
+
+```js
+it('securely edits content', () => {
+  cy.origin('supersecurelogons.com', () => {
+    cy.visit('https"//supersecurelogons.com')
+    cy.get('input#password').type('Password123!')
+    cy.get('button#submit').click()
+  })
+
+  cy.origin('mycms.com', () => {
+    cy.url().should('contain', 'cms')
+    cy.get('#current-user').contains('logged in')
+    cy.get('button#edit-1').click()
+    cy.get('input#title').type('Updated title')
+    cy.get('button#submit').click()
+    cy.get('.toast').type('Changes saved!')
+  })
+
+  cy.visit('/items/1')
+  cy.get('h1').contains('Updated title')
+})
+```
+
+Always remember,
+[Cypress tests are not unit tests](https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion).
+
+### Serialization
+
+When entering a `cy.origin()` block, Cypress injects itself at runtime, with all
+your configuration settings, into the requested origin, and sets up
+bidirectional communication with that instance. This coordination model requires
+that any data sent from one instance to another be
+[serialized](https://developer.mozilla.org/en-US/docs/Glossary/Serialization)
+for transmission. It is very important to understand that variables **inside**
+the callback are not shared with the scope **outside** the callback. For example
+this will not work:
+
+**<Icon name="exclamation-triangle" color="red"></Icon> Incorrect Usage**
+
+```js
+const foo = 1
+cy.origin('somesite.com', () => {
+  cy.visit('/')
+  // This line will throw a ReferenceError because `foo` is not defined in the
+  // scope of the callback
+  cy.get('input').type(foo)
+})
+```
+
+Instead, the variable must be explicitly passed into the callback using the
+`args` option:
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```js
+const foo = 1
+cy.origin('somesite.com', { args: { foo } }, ({ foo }) => {
+  cy.visit('/')
+  // Now it will pass
+  cy.get('input').type(foo)
+})
+```
+
+Underneath the hood, Cypress uses
+[`JSON.stringify()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)
+and
+[`JSON.parse()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse)
+(or equivalent library functions, implementation details may change) to
+serialize and deserialize the `args` object. This introduces a number of
+restrictions on the data which may be transmitted:
+
+- Primitive values only: object, array, string, number, boolean, null
+- Date objects will be converted to strings
```

Comment:
Dates are serializable by `structuredClone()`. I wonder if it's worth talking about what isn't serializable vs what is, or just linking users to [things that dont work with structured clone](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#things_that_dont_work_with_structured_clone) and covering the basics, like functions, DOM nodes, and complex object are not serializable.
```suggestion
```

Reply:
OK more types are actually allowed than I realized so I've removed the list altogether & just linked to MDN as you suggested.